### DISCUSSION TITLE
Is it possible to select text from a preview window?

### DISCUSSION THREAD

**Discussion by niba** [ibhagwan/fzf-lua#1377] [General]
In telescope it is possible to focus the preview window. I sometimes use it to yank some text without cluttering my buffers. Is it possible to do the same in fzf.lua?

**Comment by ibhagwan** [comment]
Not possible.

**Reply by zivkovicn** [reply]
Hello, is this now maybe available (couldn't find it in docs) or possible to implement? Currently in theory I can use mouse to change focus to preview window and yank some stuff from it, but would be nice if there is a keyboard shortcut to toggle the focus between search and preview window?

**Reply by ibhagwan** [reply]
Fzf is a terminal window, you can go to normal mode using `ctrl-\+ctrl-n` and then use ctrl-w… to switch to any window you want.

you can also define a bind for `focus-preview` to make it easier (in keymap.builtin).

**Reply by zivkovicn** [reply]
Thanks @ibhagwan `focus-preview` is what I needed. I was able to make it working (to have toggle between preview and search window on my key shortcut that I've define for `focus-preview`:

```lua
keymap = {
  builtin = {
    ["<C-f>"] = "focus-preview",  -- this jumps from search to preview window, where I can do whatever I want (yank, look etc.)
  },
```
then in `init` for fzf-lua I've created autocmd on `WinEnter` to find fzf window from the open ones and set the <C-f> for switching back to the search (input) window, this will also be just per buffer (local) so it doesn't bleed outside of the fzf window.

```lua
init = function()
  local group = vim.api.nvim_create_augroup("FzfLuaPreviewToggle", { clear = true })
  vim.api.nvim_create_autocmd("WinEnter", {
    group = group,
    callback = function()
      local fzf_win = vim.iter(vim.api.nvim_tabpage_list_wins(0)):find(function(w)
        return vim.bo[vim.api.nvim_win_get_buf(w)].filetype == "fzf"
      end)

      if not fzf_win or vim.bo.filetype == "fzf" then
        return
      end
      vim.keymap.set("n", "<C-f>", function()
        vim.api.nvim_set_current_win(fzf_win)
        vim.cmd("startinsert")
      end, { buffer = true, nowait = true, silent = true })
    end,
  })
end,
```
it is ugly but it works! 😄 

https://github.com/user-attachments/assets/77a765f3-9c18-423b-9ff3-a7bb5f0ce440

**Reply by nenahp** [reply]
There's hardcode one also work: https://github.com/ibhagwan/fzf-lua/blob/26dd5e39222250f751f733b1dafca9bae60cf8bb/lua/fzf-lua/previewer/builtin.lua#L201-L204

**Reply by zivkovicn** [reply]
@nenahp ha! nice, would be really great if we can pass the custom binding to it @ibhagwan  ?

**Reply by nenahp** [reply]
I guess this may be a bit overkill for this use case. A more general idea is a "preview" hook. Actually there's already  https://github.com/ibhagwan/fzf-lua/blob/f92d1ea08de01455e5da1cf7ec09979ffd2f5d3e/lua/fzf-lua/win.lua#L821... so although undocumented  `WinEnter`+`w:fzf_lua_preview` is already equalient to "enter previewer".

**Reply by ibhagwan** [reply]
> @nenahp ha! nice, would be really great if we can pass the custom binding to it @ibhagwan ?

There is, as I said, use keymap.builtin to set a bind to `focus-preview` and it will work.

**Reply by zivkovicn** [reply]
yeah, that works, we were mentioning above to be able to set the keymap for the "going back to the search, while you are focused on the preview window", currently we can do that with hardcoded letter `i` that @nenahp referred to - which is fine I guess.

**Reply by ibhagwan** [reply]
> yeah, that works, we were mentioning above to be able to set the keymap for the "going back to the search, while you are focused on the preview window", currently we can do that with hardcoded letter `i` that @nenahp referred to - which is fine I guess.

Oh yes the i, it makes logical sense IMHO also other pickers have a similar binding so users are used to it.
